external/vcm/vcm/data_transform.py

```diff
@@ -0,0 +1,103 @@
+import dataclasses
+from toolz.functoolz import curry
+from typing import Any, Callable, Literal, Mapping, MutableMapping, Sequence, Set
+import xarray as xr
+import vcm
+
+
+DATA_TRANSFORM_REGISTRY: MutableMapping[str, Mapping[str, Any]] = {}
+
+TransformName = Literal[
+    "q1_from_qm_q2", "qm_from_q1_q2", "q1_from_dQ1_pQ1", "q2_from_dQ2_pQ2",
+]
+
+
+@curry
+def register(
+    name: str,
+    inputs: Sequence[str],
+    outputs: Sequence[str],
+    func: Callable[..., xr.Dataset],
+):
+    if name in DATA_TRANSFORM_REGISTRY:
+        raise ValueError(f"Function {name} has already been added to registry.")
+    DATA_TRANSFORM_REGISTRY[name] = {
+        "func": func,
+        "inputs": inputs,
+        "outputs": outputs,
+    }
+    return func
+
+
+@register("qm_from_q1_q2", ["Q1", "Q2"], ["Qm"])
+def qm_from_q1_q2(ds):
+    ds["Qm"] = vcm.moist_static_energy_tendency(ds["Q1"], ds["Q2"])
+    return ds
+
+
+@register("q1_from_qm_q2", ["Qm", "Q2"], ["Q1"])
+def q1_from_qm_q2(ds):
+    ds["Q1"] = vcm.temperature_tendency(ds["Qm"], ds["Q2"])
+    return ds
+
+
+@register("q1_from_dQ1_pQ1", ["dQ1", "pQ1"], ["Q1"])
+def q1_from_dQ1_pQ1(ds):
+    ds["Q1"] = ds["dQ1"] + ds["pQ1"]
+    return ds
+
+
+@register("q2_from_dQ2_pQ2", ["dQ2", "pQ2"], ["Q2"])
+def q2_from_dQ2_pQ2(ds):
+    ds["Q2"] = ds["dQ2"] + ds["pQ2"]
+    return ds
+
+
+@dataclasses.dataclass
+class DataTransformConfig:
+    name: TransformName
+    kwargs: dict = dataclasses.field(default_factory=dict)
```

Comment:
What is an example of a transform that would take kwargs?

Reply:
Good question. I had some use cases but then realized they would be better suited to just defining a new transform. I still think it's possible we will have a case (e.g. thresholding at a certain value/percentile) but don't have an immediate need for the feature.

Reply:
Well, we might want to have an option of whether or not to rectify the implied surface precip to be positive when I implement the flux-form. (We could just always do it, but it might be interesting to see what happens if we don't).